rp2040-hal/src/pio.rs

```diff
@@ -1407,6 +1505,10 @@ impl<P: PIOExt> Interrupt<P> {
     ///
     /// Each of the 4 state machines have their own TX FIFO. This interrupt is raised when the TX FIFO is not full, i.e.
     /// one could push more data to it.
+    #[deprecated(
+        since = "0.7.0",
+        note = "Use the dedicated method on the state machine"
+    )]
     pub fn enable_tx_not_full_interrupt(&self, id: u8) {
         match id {
             0 => self.irq().irq_inte.modify(|_, w| w.sm0_txnfull().set_bit()),
```

Comment:
Not new in this PR, but I just noticed it:

I think this doesn't fulfill the safety contract stated above:
```
// Safety: `StateMachine` is marked Send so ensure all accesses remain atomic and no new concurrent
// accesses are added.
```
`irq_inte.modify(...)` is not atomic. Would it be possible to use the atomic bit operations instead, like you did in you new implementations?

Reply:
🤦   I had the fix committed (4min before your review) but didn't push it 🤦